### DISCUSSION TITLE
Publicly Exposed IP

### DISCUSSION THREAD

**Discussion by fredrousseau** [bionic-gpt/bionic-gpt#458] [General]
Hi Guys,

I need some support with Bionic-Gpt and/or K3S config. Thanks to the online doc, I managed to have my bionic correctly deployed in a VM. This VM is bound to a public IP ( I am using an AWS like kind of deployment) , when I try to access the webui (using my public IP ) the server controller sends back a an URL that contains the Private IP of my VM hence it fails to resolve it ... What's the best way to setup bionic /k3s to be aware I am using a public ip ?

**Comment by ianpurton** [comment]
You'll need to use the ` --hostname-url ` option on installation.

This tells Keycloak where to redirect the user to.

**Comment by ianpurton** [comment]
Also, care to share your installation method, I think it would be useful for over people.

There is some documentation around installing on AWS using EKS. https://bionic-gpt.com/docs/enterprise-edition/aws/

As I'm not familiar with AWS I'd be interested to know how you get the external IP working.

Thanks.
